index.js

```diff
@@ -460,3 +439,94 @@ function wrapPromise() {
     }
   }
 }
+
+// Shim activator for functions that have callback last
+function activator(fn) {
+  var fallback = function () {
+    var index = arguments.length - 1;
+    if (typeof arguments[index] === "function") {
+      arguments[index] = wrapCallback(arguments[index]);
+    }
+    return fn.apply(this, arguments);
+  };
+  // Preserve function length for small arg count functions.
+  switch (fn.length) {
+    case 1:
+      return function (cb) {
+        if (arguments.length !== 1) return fallback.apply(this, arguments);
+        if (typeof cb === "function") cb = wrapCallback(cb);
```

Comment:
It appears that reassigning argument variables is only a performance concern in non-strict mode. See: https://github.com/petkaantonov/bluebird/wiki/Optimization-killers#3-managing-arguments